gr-filter/lib/freq_xlating_fir_filter_impl.cc

```diff
@@ -175,10 +176,12 @@ int freq_xlating_fir_filter_impl<IN_T, OUT_T, TAP_T>::work(
 
     unsigned j = 0;
     for (int i = 0; i < noutput_items; i++) {
-        out[i] = d_r.rotate(d_composite_fir->filter(&in[j]));
-        j += this->decimation();
+        out[i] = d_composite_fir->filter(&in[j]);
+        j = j + d_decim;
```

Comment:
I would prefer using `j +=` here. WIth optimization the resulting code should be the same. Using `+=` is just cleaner IMHO.